hack/buildenv.sh

```diff
@@ -51,10 +52,23 @@ function docker_buildenv {
     opts+=(-v "${VPP_SRC}:/vpp-src")
   fi
 
-  docker run --rm --name vpp-build-${BUILD_TYPE} --shm-size 1024m \
-         ${priv} \
-         -v $PWD:/src:delegated -v $PWD/vpp-out:/vpp-out \
-         "${opts[@]}" -w /src "${DEV_IMAGE}" "$@"
+  if [[ ${DEVENV_BG} ]]; then
+    # try to kill previous background devenv container
+    docker kill $(docker ps -aq --filter name=vpp-build-${BUILD_TYPE}-bg) &>/dev/null || true
+    docker container wait $(docker ps -aq --filter name=vpp-build-${BUILD_TYPE}-bg) &>/dev/null || true
+    # ugly workaround as docker sometimes ends before the container is removed
+    sleep 1
+
+    docker run -d --rm --name vpp-build-${BUILD_TYPE}-bg --shm-size 1024m \
+          ${priv} \
+          -v $PWD:/src:delegated -v $PWD/vpp-out:/vpp-out \
+          "${opts[@]}" -w /src "${DEV_IMAGE}"
+  else
+    docker run --rm --name vpp-build-${BUILD_TYPE} --shm-size 1024m \
+          ${priv} \
+          -v $PWD:/src:delegated -v $PWD/vpp-out:/vpp-out \
+          "${opts[@]}" -w /src "${DEV_IMAGE}" "$@"
+  fi
```

Comment:
I'd rewrite this as follows. I don't think that you need to wait after `docker rm -f`, which kills a running container and then removes it. This is worth doing before any such `docker run`, not only the background one, otherwise it will fail if the container is there.
```suggestion
  name="vpp-build-${BUILD_TYPE}"
  if [[ ${DEVENV_BG} ]]; then
    name="vpp-build-${BUILD_TYPE}-bg"
    opts+=(-d)
  fi
  docker rm -f "${name}"
  docker run --rm --name "${name}" --shm-size 1024m \
         ${priv} \
         -v $PWD:/src:delegated -v $PWD/vpp-out:/vpp-out \
         "${opts[@]}" -w /src "${DEV_IMAGE}" "$@"
```

Reply:
damn, I did overcomplicate it haha